src/lib.rs

```diff
@@ -198,12 +210,30 @@ pub struct OpenzwaveAdapter {
     watchers: Arc<Mutex<Watchers>>,
 }
 
+fn ensure_directory<T: AsRef<Path> + ?Sized>(directory: &T) -> Result<(), Error> {
+    let path = directory.as_ref();
+    if path.exists() && !path.is_dir() {
+        return Err(
+            Error::IOError(io::Error::new(io::ErrorKind::AlreadyExists, format!("The file {} already exists and isn't a directory.", path.display())))
+        );
+    }
+
+    if !path.exists() {
+        try!(fs::create_dir(path));
+    }
+
+    Ok(())
+}
+
 impl OpenzwaveAdapter {
-    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>) -> Result<(), Error> {
+    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>, config_path: &str) -> Result<(), Error> {
+
+        try!(ensure_directory(config_path));
+
         let options = InitOptions {
             device: None, // TODO we should expose this as a Value
-            config_path: "./config/openzwave/",
-            user_path: "./config/openzwave/",
+            config_path: ConfigPath::Default,
+            user_path: config_path,
```

Comment:
Also, we could use some comments about what `config_path` and `user_path` are used for.